tests/models/test_renderedtifields.py

```diff
@@ -172,7 +172,10 @@ def test_delete_old_records(self, rtif_num, num_to_keep, remaining_rtifs, expect
         assert rtif_num == len(result)
 
         # Verify old records are deleted and only 'num_to_keep' records are kept
-        with assert_queries_count(expected_query_count):
+        expected_query_count_based_on_db = (expected_query_count + 1, expected_query_count)[
+            session.bind.dialect.name in ["postgresql", "sqlite", "mysql"] or expected_query_count == 0
+        ]
```

Comment:
Why does this query count change for mssql over the other backends DBs?

Reply:
we fire an extra query here https://github.com/apache/airflow/blob/4e362c134702dca30d1bd2e3da38bc2ee39825bf/airflow/models/renderedtifields.py#L169